### PR TITLE
dropdown: get rid of wrapping root `div` node (close #1392)

### LINKED ISSUE #1392: dropdown: get rid of wrapping root `div` node
Now `dropdown` has wrapping `div` that changes display mode of it's switcher (e.g. `dropdown_switcher_link` becomes `display: block` although it's based on `link` block, that is `display: inline`). I propose to get rid of this root node, make `dropdown` as mix to switcher and popup and link it by `id` js-parameter, so we can use `dropdown` as inline element without need of writing additional styles for it. To be more exact, `dropdown` will inherit display mode of it's switcher after this change.

### PR DESCRIPTION
**Pull request by aristov**
Resolves #1392

### CODE REVIEW

**Comment by tadatuta** [comment]
why not to generate `id` automatically by default?

**Comment by aristov** [comment]
@tadatuta I generate `id` automatically in template, but if I don't set `id` manually in tmpl-specs, then I get difference in `data-bem` attribute, because templaters generate a new `id` each time.

**Comment by tadatuta** [comment]
:+1:

**Comment by aristov** [comment]
@tadatuta maybe we can consider this case in `html-differ` and ignore difference for `id` param in `data-bem` attribute.

**Comment by tadatuta** [comment]
@aristov not sure if it's possible as we need to check that `id` is the same for two corresponding DOM nodes.

**Comment by narqo** [comment]
1) use `elem` instead of custom `mode`s in BEMHTML as it was before
2) add tests for proper destruct of `popup`

File: common.blocks/dropdown/dropdown.bemhtml [hunk 1]
```diff
@@ -1,8 +1,19 @@
 block('dropdown')(
-    js()(true),
+    def()(function() {
+        this.ctx.js = this.extend({ id : this.generateId() }, this.ctx.js);
+        applyCtx([apply('switcher'), apply('popup')]);
```
Comment (narqo) [inline common.blocks/dropdown/dropdown.bemhtml]:
``` js
applyCtx([
  {
    elem : 'switcher',
    js : this.extend(apply('js'), this.ctx.js),
    ···
  },
  { elem : 'popup' }
])
```

File: common.blocks/dropdown/_switcher/dropdown_switcher_link.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ modules.define('dropdown', ['link'], function(provide, _, Dropdown) {
  */
 provide(Dropdown.decl({ modName : 'switcher', modVal : 'link' }, null, /** @lends dropdown */{
     live : function() {
-        this.liveInitOnBlockInsideEvent('click', 'link', this.onSwitcherClick);
+        this.liveInitOnBlockEvent('click', 'link', this.onSwitcherClick);
```
Comment (narqo) [inline common.blocks/dropdown/_switcher/dropdown_switcher_link.js:14]: Think about moving `onSwitcherClick` to prototype.

Comment (narqo) [inline common.blocks/dropdown/_switcher/dropdown_switcher_link.js:14]: This part is not that obvious. As far as I remember it was made this way not but the mistake. One must check how clear this API will be from the pov of block inheritance. E.g. what will happen if someone will redefine `onSwitcherClick` within modifier without calling base?

Comment (aristov) [inline common.blocks/dropdown/_switcher/dropdown_switcher_link.js:14]: cc @dfilatov

Comment (aristov) [inline common.blocks/dropdown/_switcher/dropdown_switcher_link.js:14]: I propose to think about it separately from this issue.

Comment (deeonis) [inline common.blocks/dropdown/_switcher/dropdown_switcher_link.js:14]: cc @dfilatov

Comment (aristov) [inline common.blocks/dropdown/_switcher/dropdown_switcher_link.js:14]: Considered to do #1502 in `v3`.

**Comment by aristov** [comment]
TODO: ensure that dropdowns mixes work

**Comment by aristov** [comment]
TODO: ensure that `popup`'s and `dropdown__switcher`'s mixes work

**Comment by aristov** [comment]
For me: add `return` before `apply*`, squash and merge.

**Pull request merged**